lib/pgpm/package/dependencies.rb

```diff
@@ -3,14 +3,35 @@
 module Pgpm
   class Package
     module Dependencies
-      def build_dependencies
-        return ["gcc"] if c_files_present?
 
-        []
+      attr_accessor :postgres_major_version
+
+      def build_dependencies
+        case Pgpm::OS.in_scope.class.name
+        when "debian", "ubuntu"
+          [
+            "build-essential",
+            "postgresql-#{postgres_major_version}",
```

Comment:
Postgres dependencies are added in the RPM spec / DEB control preparation based on `Distribution`'s methods.